### ISSUE TITLE
Auth Trigger isn't config in Cognito User Pool when amplify push

### ISSUE THREAD

**Issue by phanviet** [aws-amplify/amplify-cli#7561]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm i -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

v12.21.0

### Amplify CLI Version

5.0.1

### What operating system are you using?

Mac

### Amplify Categories

auth

### Amplify Commands

push

### Describe the bug

I setup Cognito user pool via CLI, config basic information + OAuth (Facebook) + choose 3 Lambda Trigger: PostAuthentication, PostConfirmation, PreSignup, completed all steps without any errors or warnings. 

When I call `amplify push`, Access Cognito User Pool -> Choose Triggers -> Don't see any lambda trigger. 3 Lambda deployed but not linked to Cognito User Pool Triggers.
The Cloudformation template file `auth-trigger-cloudformation-template.json` generated but never deployed.

### Expected behavior

Cognito Lambda Trigger should be linked to Cognito User Pool Triggers

### Reproduction steps

I setup Cognito user pool via CLI, config basic information + OAuth (Facebook) + choose 3 Lambda Trigger: PostAuthentication, PostConfirmation, PreSignup, completed all steps without any errors or warnings. 

When I call `amplify push`, Access Cognito User Pool -> Choose Triggers -> Don't see any lambda trigger. 3 Lambda deployed but not linked to Cognito User Pool Triggers.
The Cloudformation template file `auth-trigger-cloudformation-template.json` generated but never deployed.

**Comment by attilah** [comment]
@phanviet I was trying to reproduce the error that you describing but I was not able to do as the triggers were associated with the userpool I configured:

![image](https://user-images.githubusercontent.com/230432/122841053-34645600-d2b0-11eb-9b61-ca4555626c53.png)

> The Cloudformation template file auth-trigger-cloudformation-template.json generated but never deployed.

I assume you are using VS Code as your editor and by default for Amplify project we are hiding certain files root stack is one of them. If you look into `amplify/backend/awscloudformatin/nested-cloudformation-stack.yml` file you can locate this section:

```json
    "AuthTriggerCustomLambdaStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": "https://s3.amazonaws.com/amplify-yyy-dev-162550-deployment/amplify-cfn-templates/auth/auth-trigger-cloudformation-template.json",
```

To be able to reproduce the error please send me the full text of the walkthrough, so I could try it again.

**Comment by phanviet** [comment]
@attilah thx, I also don't see the folder `amplify/backend/awscloudformation` and `amplify-meta.json`. I'm using Amplify CI/CD to trigger build when I push code, and use `amplify env pull` to sync the changed from cloud to dev team. Are they difference?

I tried to `amplify init`, `amplify push` at local but no luck, anything else I can do to have these without remove stack and init again?

**Comment by phanviet** [comment]
@attilah I also don't see any files in `amplify/amplify`, just 3 empty folders `.config`, `#current-cloud-backend`, `backend`.

**Comment by phanviet** [comment]
@attilah I knew the cause I don't have these files

When I `amplify init` with default config (javascript app), it's fine and I have `amplify-meta.json` as well as `#current-cloud-backend`.

But when I `amplify init` with react app, I don't have `amplify-meta.json`, and folders `.config`, `#current-cloud-backend`, `backend` are empty.

Is that bug?

**Comment by phanviet** [comment]
Updated: although I have this file `amplify/backend/awscloudformation/nested-cloudformation-stack.yml` after `amplify push`, but I don't see the nested stack of `auth-trigger-cloudformation-template.json`

Below is my detail when `amplify add auth`

![Screen Shot 2021-06-22 at 13 02 23](https://user-images.githubusercontent.com/2238528/122872543-cfb20580-d35a-11eb-8090-28fefc13e3c5.png)

**Comment by phanviet** [comment]
@attilah 

I think this bug happens when I add more User pool Group, please help reproduce as below screenshot:
<img width="1065" alt="Screen Shot 2021-06-22 at 13 53 52" src="https://user-images.githubusercontent.com/2238528/122878705-3e469180-d362-11eb-8c41-7eda8643cdb1.png">

**Comment by BeaveArony** [comment]
I also have this problem after updating to amplify cli 5.0.2. I first got a circular dependency error involving the AdminQueries API. Then I found the feature flag:
```json
{
  "features": {
    "auth": {
      "breakCircularDependency": true
    }
  }
}
``` 
It solved the circular dependency error by removing the triggers from the auth *-cloudformation-template.yml and adding a `auth-trigger-cloudformation-template.json` file with the triggers in it. When I run `amplify push` it does NOT result in the "AuthTriggerCustomLambdaStack" resource being created in `amplify/backend/awscloudformation/nested-cloudformation-stack.yml`. 

I did the same in an old test-project that had triggers in it, upgraded to amplify CLI 5.0, added the feature flag and there everything works fine. I see the "AuthTriggerCustomLambdaStack" section and the triggers appear after `amplify push`.

How can I fix this in the other (production) project?
